### PR TITLE
S157 multi backend

### PR DESCRIPTION
**Pull request by slfritchie**
- Add multi backend source that has a bucket name prefix matching feature
- Add README.multi-backend.txt file with (still buggy) instructions for configuring the multi backend for Riak
- Add the still-unfinished fs2 backend, based originally on the Riak 0.14 riak_fs_backend.erl and updated
  enough capability to put & get objects and to use on-disk checksums.

### CODE REVIEW

File: README.multi-backend.txt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Step a: Identify the directory that will be storing Riak's data.
+
+            Riak Linux data directory path:
+                /var/lib/riak
+
+            Riak Solaris data directory path:
+                /opt/riak/data
+
+            Riak developer source data directory path:
+                ./rel/riak/data
```
Comment (reiddraper) [inline README.multi-backend.txt]: I _think_ this should just be `./data`, as the path will be relative from the app.config location.

Comment (reiddraper) [inline README.multi-backend.txt]: Rather, when you refer to it later as the substitution for DATADIR.

Comment (slfritchie) [inline README.multi-backend.txt]: Yup, you're right.  I'll fix it.

**Comment by reiddraper** [comment]
I wasn't sure if there was a better way to test, but I've been able to confirm that the Right Backend is used by just deleting the whole _other_ data dir and having data still persist. Looks good to me, +1. Nice clear instructions too, btw.

**Pull request merged**